core/src/main/java/com/datastax/oss/driver/api/core/type/codec/TypeCodec.java

```diff
@@ -86,7 +86,7 @@ default boolean accepts(Class<?> javaClass) {
    */
   default boolean accepts(Object value) {
     Preconditions.checkNotNull(value);
-    return getJavaType().__getToken().isSupertypeOf(TypeToken.of(value.getClass()));
+    return getJavaType().isSupertypeOf(GenericType.of(value.getClass()));
```

Comment:
👍 for isolating that in separate commits, that makes it easy to squash with the "Enhance GenericType..." one.